### PR TITLE
Add configurations for formatting options

### PR DESCRIPTION
**Pull request by sanjana**
- `Tab Size` - numeric type - default: 4
- `Column Limit` - numeric type - default: 120
- `Whitespace Character` - string type - default: ` `

Related Issue: https://github.com/ballerina-platform/ballerina-lang/issues/28234

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -62,6 +62,21 @@
             "type": "object",
             "title": "Ballerina",
             "properties": {
+                "ballerina.formattingOptions.tabsize": {
+                    "type": "number",
+                    "default": 4,
+                    "description": "Specifies the tab size when formatting ballerina code (Supports for Swan-Lake-Alpha-2 onwards)"
```
Comment (rasika) [inline package.json:68]: Need to remove `-Alpha-2`, Shall we rephrase this as `Supports Swan-Lake onwards`?

**Review by rasika — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (IrushiL) [reply package.json:68]: A suggestion for the wording:
```suggestion
                    "description": "Specifies the tab size when formatting ballerina code (supported on versions following Ballerina Swan Lake)"
```

File: package.json [hunk 2]
```diff
@@ -62,6 +62,21 @@
...
+                },
+                "ballerina.formattingOptions.columnLimit": {
+                    "type": "number",
+                    "default": 120,
+                    "description": "Specifies the column limit when formatting ballerina code (Supports for Swan-Lake-Alpha-2 onwards)"
```
Comment (prabushi) [inline package.json:73]:
```suggestion
                    "description": "Specify the line character limit for formatter.\n(Supports Ballerina Swan Lake)"
```

**Comment by anupama-pathirage** [comment]
@sanjana Is  this PR still valid? It seems we don't have any recent update on this and becoming old to keep open.

**Comment by sanjana** [comment]
> @sanjana Is this PR still valid? It seems we don't have any recent update on this and becoming old to keep open.

@anupama-pathirage since there're yet unresolved discussions regarding this PR on [Github Discussions](https://github.com/ballerina-platform/ballerina-lang/discussions/28641) and [Slack](https://ballerina-platform.slack.com/archives/CAYCPFRHQ/p1613629863067600) it was converted to a draft state. 

Would it be better if this is closed for the moment?

**Comment by sanjana** [comment]
Closing since not decided to proceed further with this feature at the moment in SL releases.
